core/src/main/java/com/scalar/db/storage/objectstorage/MutateStatementHandler.java

```diff
@@ -0,0 +1,226 @@
+        getTable(mutation),
+        objectStorageMutation.getConcatenatedPartitionKey(),
+        Collections.singletonList(mutation));
+  }
+
+  public void handle(List<? extends Mutation> mutations) throws ExecutionException {
+    Map<PartitionIdentifier, List<Mutation>> mutationPerPartition = new HashMap<>();
+    for (Mutation mutation : mutations) {
+      TableMetadata tableMetadata = metadataManager.getTableMetadata(mutation);
+      ObjectStorageMutation objectStorageMutation =
+          new ObjectStorageMutation(mutation, tableMetadata);
+      String partitionKey = objectStorageMutation.getConcatenatedPartitionKey();
+      PartitionIdentifier partitionIdentifier =
+          PartitionIdentifier.of(getNamespace(mutation), getTable(mutation), partitionKey);
+      mutationPerPartition
+          .computeIfAbsent(partitionIdentifier, k -> new ArrayList<>())
+          .add(mutation);
+    }
+    for (Map.Entry<PartitionIdentifier, List<Mutation>> entry : mutationPerPartition.entrySet()) {
+      PartitionIdentifier partitionIdentifier = entry.getKey();
+      mutate(
+          partitionIdentifier.getNamespaceName(),
+          partitionIdentifier.getTableName(),
+          partitionIdentifier.getPartitionName(),
+          entry.getValue());
+    }
+  }
+
+  private void mutate(
+      String namespaceName, String tableName, String partitionKey, List<Mutation> mutations)
+      throws ExecutionException {
+    Map<PartitionIdentifier, String> readVersionMap = new HashMap<>();
+    ObjectStoragePartition partition =
+        getPartition(namespaceName, tableName, partitionKey, readVersionMap);
+    for (Mutation mutation : mutations) {
+      if (mutation instanceof Put) {
+        partition.applyPut((Put) mutation, metadataManager.getTableMetadata(mutation));
+      } else {
+        assert mutation instanceof Delete;
+        partition.applyDelete((Delete) mutation, metadataManager.getTableMetadata(mutation));
+      }
+    }
+    applyPartitionWrite(namespaceName, tableName, partitionKey, partition, readVersionMap);
+  }
+
+  /**
+   * Applies the partition write.
+   *
+   * @param namespaceName the namespace name
+   * @param tableName the table name
+   * @param partitionKey the partition key
+   * @param partition the partition to be written
+   * @param readVersionMap the map of read versions
+   * @throws ExecutionException if a failure occurs during the operation
+   */
+  private void applyPartitionWrite(
+      String namespaceName,
+      String tableName,
+      String partitionKey,
+      ObjectStoragePartition partition,
+      Map<PartitionIdentifier, String> readVersionMap)
+      throws ExecutionException {
+    if (readVersionMap.containsKey(partition.getPartitionIdentifier())) {
+      String readVersion = readVersionMap.get(partition.getPartitionIdentifier());
+      if (!partition.isEmpty()) {
+        updatePartition(namespaceName, tableName, partitionKey, partition, readVersion);
```

Comment:
Can these be like this?
```suggestion
        updatePartition(partition, readVersion);
```